### PR TITLE
Saturday Edition & Updated Registration Consents/Newsletter Flow

### PR DESCRIPTION
**Pull request by coldlink**
### Commit by commit
- Add ability to update newsletters from "registrationConsents"
- Update the `ToggleSwitch` component to make it more flexible, i.e allow context and image
- Add the `RegistrationNewsletterFormField` component, and style it to be like the Saturday Edition design
- Add it to the `RegisterWithEmail` and `WelcomeSocial` pages
- Use geolocation to show/hide the Saturday Edition component
  - Show for UK/EU/ROW
  - Hide for AU/US
- Make the registration newsletter and consents components more generic so they're easier to update in the future to change/add/remove things
- Add the cypress tests for Saturday edition
  - Since the CODE newsletters api isn't up to date with prod, we use a workaround when testing
  - When the registration newsletter/consent is updated in IDAPI, we set a cypress only cookie called `cypress-newsletter-response` and `cypress-consent-response` which contains the JSON we sent to IDAPI
  - If this matches what we initially sent at the start of the flow, we can be reasonably sure that the newsletter/consent was updated, even if the API itself isn't up to date
- Fix: Make sure consents/newsletters are updated on the `POST /welcome/:social` call
- Fix: Make sure Saturday Edition appears on Onboarding review page if selected

#### GB/EU/ROW

<table>
<tr>
 <td>
  Mobile
 <td>
  Desktop
<tr>
 <td>

60929d168594f80039336501-bnsrfxbqck chromatic com_iframe html_args= id=pages-registerwithemail--default viewMode=story(iPhone 12 Pro)

 <td>

60929d168594f80039336501-bnsrfxbqck chromatic com_iframe html_args= id=pages-registerwithemail--default viewMode=story(Nest Hub Max)

<tr>
<td>

60929d168594f80039336501-bnsrfxbqck chromatic com_iframe html_args= id=pages-welcomesocial--google viewMode=story(iPhone 12 Pro)
<td>

60929d168594f80039336501-bnsrfxbqck chromatic com_iframe html_args= id=pages-welcomesocial--google viewMode=story(Nest Hub Max)

</table>

#### US/AU

<table>
<tr>
 <td>
  Mobile
 <td>
  Desktop
<tr>
 <td>

60929d168594f80039336501-bnsrfxbqck chromatic com_iframe html_args= id=pages-registerwithemail--us-geolocation viewMode=story(iPhone 12 Pro) (1)

 <td>

60929d168594f80039336501-bnsrfxbqck chromatic com_iframe html_args= id=pages-registerwithemail--us viewMode=story(Nest Hub Max)

<tr>

<td>

60929d168594f80039336501-bnsrfxbqck chromatic com_iframe html_args= id=pages-welcomesocial--us-geolocation viewMode=story(Nest Hub Max)

<td>

60929d168594f80039336501-bnsrfxbqck chromatic com_iframe html_args= id=pages-welcomesocial--us-geolocation viewMode=story(iPhone 12 Pro)

</table>

## Tested
- [x] CODE GB Geolocation (check if IDAPI call succeeds)
- [x] CODE US Geolocation - newsletter should not appear

### CODE REVIEW

**Review by kelvin-chappell — DISMISSED** [review]
Makes sense to me 👍

File: cypress/integration/ete-okta/onboarding_flow.5.cy.ts [hunk 1]
```diff
@@ -261,7 +261,7 @@ describe('Onboarding flow', () => {
 // These tests checks this behaviour by first getting a user into this state
 // i.e user.profile.emailValidated = false, and user groups has GuardianUser-EmailValidated
 describe('Social Registration - Consents Page', () => {
-	it('shows the consents page for social registration using google - opts in', () => {
+	it.only('shows the consents page for social registration using google - opts in', () => {
```
Comment (raphaelkabo) [inline cypress/integration/ete-okta/onboarding_flow.5.cy.ts:264]: ❗ 😡

**Review by raphaelkabo — COMMENTED** [review]
So close.

File: src/client/pages/WelcomeSocial.tsx [hunk 2]
```diff
@@ -64,19 +68,24 @@ const inlineMessage = (socialProvider: SocialProvider) => css`
...
+	geolocation,
 }: WelcomeSocialProps) => {
 	const formTrackingName = 'register';
 
 	usePageLoadOphanInteraction(formTrackingName);
 
+	// don't show the Saturday Edition newsletter option for US and AU
+	const showSaturdayEdition = !(['US', 'AU'] as GeoLocation[]).some(
+		(location: GeoLocation) => location === geolocation,
+	);
```
Comment (kelvin-chappell) [inline src/client/pages/WelcomeSocial.tsx:87]: Is it just US and AU that are excluded or everywhere except GB?

**Review by kelvin-chappell — DISMISSED** [review]
👍

File: src/client/pages/WelcomeSocial.tsx (hunk 2 above)
Comment (coldlink) [reply src/client/pages/WelcomeSocial.tsx:87]: Just US and AU

**Review by raphaelkabo — APPROVED** [review]
👀

**Pull request merged**